packages/keyring-controller/src/KeyringController.ts

```diff
@@ -467,7 +477,11 @@ export class KeyringController extends BaseControllerV2<
   async addNewKeyring(
     type: KeyringTypes | string,
     opts?: unknown,
-  ): Promise<Keyring<Json>> {
+  ): Promise<unknown> {
```

Comment:
Not strictly required by this PR, but makes more sense since the keyring created could be of any type (for example, a `QRKeyring`)

Reply:
Can we have a base class/interface for the Keyring, then having a child classes with specific Keyring types for example QRKering.

Reply:
We have a type that should be implemented by all Keyrings: https://github.com/MetaMask/utils/blob/4589c755e3005024c9071dd0c23c933335c3efa4/src/keyring.ts#L47

But that's not yet the case, so we still have to handle their API differences. In this specific case, it is more correct to say that we don't know what we are returning, instead of assuming that we are returning a `Keyring<Json` - also because we don't restrict the type of keyrings that can be created through this controller